main/src/library/File.flix

```diff
@@ -59,4 +159,180 @@ namespace File {
         else
             readAllHelper(r, line :: acc)
 
+    ///
+    /// Returns a string containing the given file `f`.
+    ///
+    pub def read(f: String): Result[String, String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path);
+            import java.io.BufferedReader.close();
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath);
+            let result = readHelper(readAll(reader));
+            close(reader);
+            Ok(result)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Helper function for read.
+    /// Converts a list of strings to a string with linebreaks between every string in `list`.
+    ///
+    def readHelper(list: List[String]): String & Pure =
+        List.foldLeft((x, y) -> x + "\n" + y, "", list)
+
+    ///
+    /// Returns a string containing the given file `f` with the options `opts`.
+    ///
+    pub def readWith(opts: {offset :: Int64, count :: Int32, charSet :: String}, f : String): Result[String, String] & Impure = {
+        try {
+            import new java.io.File(String) as newFile;
+            import java.nio.charset.Charset:forName(String);
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path, ##java.nio.charset.Charset);
+            import java.io.BufferedReader.close();
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath, forName(opts.charSet));
+            let fileRead = readHelper(readAll(reader));
```

Comment:
I don't know how much efficiency is a concern, but it seems wasteful to have the whole file in memory to read a potentially small bit of information

Reply:
What do you mean? We have to read the entire file to return it as a string, right? :)

Reply:
Oooh. We are not taking the offset into acocunt. Yes, we should do that, i.e. not read the entire thing. Ignore my previous comment.